components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2154,13 +2162,36 @@ private static void updateCookieConfig(CookieBuilder cookieBuilder, IdentityCook
     public static String getMultiAttributeSeparator() {
 
         String multiAttributeSeparator = null;
-        try {
-            multiAttributeSeparator = CarbonContext.getThreadLocalCarbonContext().getUserRealm().
-                    getRealmConfiguration().getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR);
-        } catch (UserStoreException e) {
-            log.warn("Error while retrieving MultiAttributeSeparator from UserRealm.");
-            if (log.isDebugEnabled()) {
-                log.debug("Error while retrieving MultiAttributeSeparator from UserRealm.", e);
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_ENABLED))) {
+            try {
+                Attribute configAttribute = FrameworkServiceDataHolder.getInstance().getConfigurationManager()
+                        .getAttribute(MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_RESOURCE_TYPE,
+                                MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_RESOURCE_NAME,
+                                MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_ATTRIBUTE_NAME);
+
+                if (configAttribute != null && StringUtils.isNotBlank(configAttribute.getValue())) {
+                    multiAttributeSeparator = configAttribute.getValue();
+                }
```

Comment:
if (configAttribute == null ), let's set the file-based attribute. Otherwise, if it sets to null in an unexpected scenario there can be issues.

Reply:
Isn't this logic correct? If the attribute value is defined and is not empty, we retrieve it. In any other flow, file based separator is used during the next string empty check.